File: service/src/main/java/com/heitor/minhaApi/security/UsuarioLoginDTO.java
package com.heitor.minhaApi.security;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UsuarioLoginDTO {

    @NotEmpty(message = "Campo obrigatório")
    private String username;
    @NotEmpty(message = "Campo obrigatório")
    @Size(min = 6, max = 20, message = "Campo deve conter de 6 a 20 caracteres")
    private String password;
}

File: client/src/models/UsuarioLoginDTO.ts
export class UsuarioLoginDTO {
    username: string
    password: string

    constructor(username: string, password: string) {
        this.username = username;
        this.password = password;
    }

    setUsername(username: string){
        this.username = username
    }

    setPassword(password: string){
        this.password = password
    }
}